tests/test_quickLook.py

```diff
@@ -0,0 +1,313 @@
+                               )
+        self.assertIsNotNone(result, "Result of run method should not be None")
+        self.assertIsInstance(result, pipeBase.Struct, "Result should be of type lsst.pipe.base.Struct")
+        self.assertIsInstance(result.exposure, afwImage.Exposure,
+                              "Resulting exposure should be an instance of lsst.afw.image.Exposure")
+
+    def test_runQuickLookMissingData(self):
+        # Test without any inputs other than the exposure
+        result = self.task.run(self.ccdExposure)
+        self.assertIsInstance(result.exposure, afwImage.Exposure)
+
+    def test_runQuickLookBadDark(self):
+        # Test with an incorrect dark frame
+        bbox = self.ccdExposure.getBBox()
+        bbox.grow(-20)
+        with self.assertRaises(lsst.pex.exceptions.wrappers.LengthError):
+            self.task.run(self.ccdExposure, camera=self.camera, bias=self.bias,
+                          dark=self.dark[bbox], flat=self.flat, defects=self.defects)
+
+
+class QuickLookIsrTaskRunQuantumTests(lsst.utils.tests.TestCase):
+    """Tests of ``QuickLookIsrTask.runQuantum``, which need a test butler,
+    but do not need real images.
+
+    Adapted from the unit tests of ``CalibrateImageTask.runQuantum``
+    """
+    def setUp(self):
+        instrument = "testCam"
+        exposureId = 100
+        visit = 100101
+        detector = 0
+        physical_filter = "testCam_filter"
+        band = "X"
+
+        # Map the isrTask connection names to the names of the Butler dataset
+        # inputs
+        ccdExposure = "raw"
+        camera = "camera"
+        bias = "bias"
+        dark = "dark"
+        flat = "flat"
+        defects = "defects"
+        bfKernel = "bfKernel"
+        newBFKernel = "brighterFatterKernel"
+        ptc = "ptc"
+        filterTransmission = "transmission_filter"
+        deferredChargeCalib = "cpCtiCalib"
+        opticsTransmission = "transmission_optics"
+        strayLightData = "yBackground"
+        atmosphereTransmission = "transmission_atmosphere"
+        crosstalk = "crosstalk"
+        illumMaskedImage = "illum"
+        linearizer = "linearizer"
+        fringes = "fringe"
+        sensorTransmission = "transmission_sensor"
+        crosstalkSources = "isrOverscanCorrected"
+
+        # outputs
+        outputExposure = "postISRCCD"
+
+        # quickLook-only outputs
+        exposure = "quickLookExp"
+
+        # Create a and populate a test butler for runQuantum tests.
+        self.repo_path = tempfile.TemporaryDirectory(ignore_cleanup_errors=True)
+        self.repo = butlerTests.makeTestRepo(self.repo_path.name)
+
+        # dataIds for fake data
+        butlerTests.addDataIdValue(self.repo, "instrument", instrument)
+        butlerTests.addDataIdValue(self.repo, "physical_filter", physical_filter, band=band)
+        butlerTests.addDataIdValue(self.repo, "detector", detector)
+        # butlerTests.addDataIdValue(self.repo, "detector", detector + 1)
```

Comment:
Should this commented line be removed? Or was the plan to iterate over this option and check it raises for a detector id mismatch?